ONSdigital/dp-api-tests
# Feature/extend dataset tests

### What

Extend test coverage for almost all endpoints on the dataset API and refactor code to use gherkin (given, when, then). Test coverage for version and instance dimensions not extended due to restructuring the endpoints, which will mean a revamp of the spec for `/datasets/...../dimensions`, `/instances/....../dimensions`, `/datasets/...../options` and `/instances/....../options`.

### How to review

Make sure you have checkout `feature/fix-dataset-state-changes` on dp-dataset-api, run the tests by doing the following:

1) Run dataset API
2) cd `<path>/dp-api-tests/datasetAPI`
3) Run `go test ./...`

Make sure tests make sense and follow spec

### Who can review

Team B or Manoj
